include/cantera/oneD/refine.h

```diff
@@ -69,51 +123,142 @@ class Domain1D;
+     * the refinement analysis.
+     */
     void show();
+
+    /**
+     * Returns true if a new grid point is needed to the right of grid index j.
+     *
+     * @param j  Grid point index
+     */
     bool newPointNeeded(size_t j) {
-        return m_loc.find(j) != m_loc.end();
+        return m_insertPts.find(j) != m_insertPts.end();
     }
+
+    /**
+     * Returns true if the grid point at index j should be kept.
+     *
+     * @param j Grid point index
+     */
     bool keepPoint(size_t j) {
-        return (m_keep[j] != -1);
+        return (m_keep[j] != REMOVE);
     }
-    double value(const double* x, size_t i, size_t j);
 
+    /**
+     * Returns the value of the solution component, n, at grid point j.
+     *
+     * @param x Solution vector
+     * @param n Solution Component index
+     * @param j Grid point index
+     */
+    double value(const double* x, size_t n, size_t j);
+
+    //! Returns the maximum ratio of grid spacing at adjacent intervals
     double maxRatio() {
         return m_ratio;
     }
+
+    //! Returns the maximum allowable difference in value between adjacent points
     double maxDelta() {
         return m_slope;
     }
+
+    //! Returns the maximum allowable difference in the derivative between adjacent points
     double maxSlope() {
         return m_curve;
     }
+
+    //! Returns the threshold for removing unnecessary grid points
     double prune() {
         return m_prune;
     }
 
 protected:
     //! Indices of grid points that need new grid points added after them
-    set<size_t> m_loc;
-    map<size_t, int> m_keep;
+    set<size_t> m_insertPts;
+
+    // Grid point refinement status
+    enum GridPointStatus {
+        UNSET = 0,
+        KEEP = 1,
+        REMOVE = -1
+    };
+
+    //! Status of whether each grid point should be kept or removed.
+    map<size_t, GridPointStatus> m_keep;
+
     //! Names of components that require the addition of new grid points
-    set<string> m_c;
+    set<string> m_componentNames;
+
+    //! Flags for whether each component should be considered for grid refinement
     vector<bool> m_active;
-    double m_ratio = 10.0;
-    double m_slope = 0.8;
-    double m_curve = 0.8;
-    double m_prune = -0.001;
+
+    //! Refinement criteria
+    double m_ratio = 10.0; //!< grid spacing refinement criteria
+    double m_slope = 0.8; //!< function change refinement criteria
+    double m_curve = 0.8; //!< function slope refinement criteria
+    double m_prune = -0.001; //!< pruning refinement criteria
+
+    //! Threshold for ignoring small changes around a constant during refinement.
     double m_min_range = 0.01;
-    Domain1D* m_domain;
-    size_t m_nv;
-    size_t m_npmax = 1000;
+
+    Domain1D* m_domain; //!< Pointer to the domain to be refined.
+    size_t m_nv; //!< Number of components in the domain
+    size_t m_npmax = 1000; //!< Maximum number of grid points
+
+    //! Closest that a solution component at two adjacent grid points can be.
```

Comment:
I think you're misinterpreting the use of `m_thresh`. Its purpose is to provide an absolute tolerance for the slope and curve evaluations such that solution components whose values don't change very much across the domain aren't targeted for refinement if there happens to be a small blip at some point in the domain. It's very much analogous to the absolute tolerances used in the test suite comparisons.

Reply:
That's my understanding I believe. Components that are closer than the m_thresh hit the m_thresh floor with the way it is used in the implementation. The slope criteria uses it in a derived way to limit the slope based on the solution component values versus as a floor of the slope. ( the use of the m_thresh/dz in the implementation instead of just using m_thresh.

Maybe a longer comment of:

//! Absolute tolerance for solution components in the domain to prevent the refiner from considering numerical noise